documentation/service_onboarding.md

```diff
@@ -0,0 +1,164 @@
+```golang
+// Fprint formats using the default formats for its operands and writes to w.
+// Spaces are added between operands when neither is a string.
+// It returns the number of bytes written and any write error encountered.
+func Fprint(w io.Writer, a ...interface{}) (n int, err error) {
+```
+
+Each package needs to include a `doc.go` file and not be a part of a service version. For more details about this file there is a detailed write-up in the [repo wiki](https://github.com/Azure/azure-sdk-for-go/wiki/doc.go-template). In the `doc.go` file you should include a short service overview, basic examples, and (if they exist) a link to samples in the [`azure-sdk-for-go-samples` repository](https://github.com/azure-samples/azure-sdk-for-go-samples)
+
+### Constructors
+All clients should be able to be initialized directly from the user and should begin with `New`. For example to define a constructor for a new client for the Tables service we start with defining the struct `TableServiceClient`:
+```golang
+// A TableServiceClient represents a client to the table service. It can be used to query the available tables, add/remove tables, and various other service level operations.
+type TableServiceClient struct {
+	client  *tableClient
+	service *serviceClient
+	cred    SharedKeyCredential
+}
+
+```
+Note that there are no exported fields on the `TableServiceClient` struct, and as a rule of thumb, generated clients and credentials should be private.
+
+Constructors for clients are separate methods that are not associated with the struct. The constructor for the TableServiceClient is as follow:
+```golang
+// NewTableServiceClient creates a TableServiceClient struct using the specified serviceURL, credential, and options.
+func NewTableServiceClient(serviceURL string, credential azcore.Credential, options *TableServiceClientOptions) (*TableServiceClient, error) {
+	conOptions := options.getConnectionOptions()
+	if isCosmosEndpoint(serviceURL) {
+		conOptions.PerCallPolicies = []azcore.Policy{CosmosPatchTransformPolicy{}}
+	}
+	con := newConnection(serviceURL, cred, conOptions)
+	c, err := cred.(*SharedKeyCredential)
+	return &TableServiceClient{client: &tableClient{con}, service: &serviceClient{con}, cred: *c}, err
+}
+```
+In `Go`, the method parameters are enclosed with parenthesis immediately following the method name with the parameter name preceding the parameter type. The return arguments follow the parameters. If a method has more than one return parameter the types of the parameter must be enclosed in parenthesis. Note the `*` before a type indicates a pointer to that type. All methods that create a new client or interact with the service should return an `error` type as the last argument.
```

Comment:
Is the purpose of this doc to articulate our "flavor of Go" (ie, our standard and guidelines) or the literal syntax of Go itself?

Reply:
I'm trying to strike a balance on the literal syntax of `Go`. I'm making the assumption that the partner team either has never written Go or only has a beginners level knowledge.

Reply:
In JS (and probably others) there's usually a "template" project. Would you say the Tables client is ours and we should strip it down a bit so people can copy and go?

Reply:
I think adding a template project is a good idea, I'll look into this